dispatch/tests/test_api_pages.py

```diff
@@ -0,0 +1,239 @@
+
+    def test_create_empty_page(self):
+        """
+        Create page should fail with empty payload
+        """
+
+        url = reverse('api-pages-list')
+
+        response = self.client.post(url, None, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_create_incomplete_page(self):
+        """
+        Creat Page should fail with missing required fields
+        """
+
+        url = reverse('api-pages-list')
+
+        # Page missing content, snippet, slug
+        data = {
+            "title": "Test Page",
+        }
+
+        response = self.client.post(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        # Check Data
+        self.assertTrue('content_json' in response.data)
+        self.assertTrue('snippet' in response.data)
+        self.assertTrue('slug' in response.data)
+
+    def test_create_page(self):
+        """
+        Ensure we can make a page
+        """
+
+        response = self._create_page()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # Check Data
+        self.assertEqual(response.data['title'], 'Test Page')
+        self.assertEqual(response.data['slug'], 'test_page')
+        self.assertEqual(response.data['snippet'], "This is a test snippet")
+        self.assertEqual(response.data['content'][0]['type'], 'paragraph')
+        self.assertEqual(response.data['content'][0]['data'], 'This is some paragraph text')
+
+    def test_update_fields_page(self):
+        """
+        Should be able to update the title, slug, snippet of the page
+        """
+
+        # First make a page
+        page = self._create_page()
+
+        # Generate detail url
+        url = reverse('api-pages-detail', args=[page.data['id']])
+
+        # Change the fields
+        newdata = {
+          "title": "New Test Page",
+          "slug": "New test-page",
+          "snippet": "This is a new test snippet",
+          "content": [
+            {
+              "type": "paragraph",
+              "data": "This is some paragraph text"
+            }
+          ]
+        }
+
+        response = self.client.patch(url, newdata, format='json')
+
+        # Check data
+        self.assertEqual(response.data['title'], "New Test Page")
+        self.assertEqual(response.data['slug'], "New test-page")
+        self.assertEqual(response.data['snippet'], "This is a new test snippet")
+
+    def test_update_content_page(self):
+        """
+        Should be able to update contents of the page
+        """
+
+        # First make a page
+        page = self._create_page()
+
+        # Generate detail url
+        url = reverse('api-pages-detail', args=[page.data['id']])
+
+        # Change the content
+        newdata = {
+          "title": "Test Page",
+          "slug": "test-page",
+          "snippet": "This is a test snippet",
```

Comment:
The `slug` is what determines the url for article/page, and is unique to each instance.

For example, the slug for this article is `ubc-researchers-works-to-fight-fake-news`

![screen shot 2017-04-16 at 10 33 57 pm](https://cloud.githubusercontent.com/assets/2547035/25079876/ff14e678-22f4-11e7-85a1-e4ddfdaa8a9c.png)